File: src/main/java/cal/fstg/java/service/DocumentService.java
package cal.fstg.java.service;

import cal.fstg.java.dao.DocumentRepository;
import cal.fstg.java.dao.EmployerRepository;
import cal.fstg.java.model.Document;
import cal.fstg.java.model.Employer;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class DocumentService {

    private final DocumentRepository documentRepository;

    @Autowired
    public DocumentService(DocumentRepository documentRepository) {
        this.documentRepository = documentRepository;
    }

    public List<Document> getAll() {
        return documentRepository.findAll();
    }

    public Optional<Document> getOne(long id) {
        return documentRepository.findById(id);
    }

    public Document save(Document document) {
        return documentRepository.save(document);
    }

    public void delete(long id) {
        documentRepository.deleteById(id);
    }
}

File: src/main/kotlin/cal/fstg/stagerkotlin/service/DocumentService.kt
package cal.fstg.stagerkotlin.service

import cal.fstg.stagerkotlin.dao.DocumentRepository
import cal.fstg.stagerkotlin.model.Document
import kotlinx.coroutines.ExperimentalCoroutinesApi
import kotlinx.coroutines.reactor.flux
import kotlinx.coroutines.reactor.mono
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class DocumentService(@Autowired private val documentRepository: DocumentRepository) {

    @ExperimentalCoroutinesApi
    fun getAll() = flux<Document> { documentRepository.findAll() }

    fun getOne(id: String) = mono { documentRepository.findById(id) }

    fun save(document: Document) = mono { documentRepository.save(document) }

    fun delete(id: String) = documentRepository.deleteById(id)
}